### ISSUE TITLE
Add UsesXml annotation

### ISSUE THREAD

**Issue by ColinTree** [mit-cml/appinventor-sources#1245]
This may be required to create custom Resources like listview item layout.

**Comment by ewpatton** [comment]
I can see how this would be useful for extensions, but for customizing the appearance of built-in components I would prefer that we have a proper editor for letting users manage the layout of items.

**Comment by moliata** [comment]
UsesXml should also allow to add custom XML code into AndroidManifest like Compiler.java does.

**Comment by ewpatton** [comment]
Rather than introduce a `@UsesXml` annotation with overloaded semantics, I would suggest instead that we update the extension mechanism and compiler to allow for the inclusion of an AndroidManifest.xml and res/... directory structure similar to how Android Archive (.aar) files work. Then, we can have aapt handle merging these along with any .aar files and the application's AndroidManifest.xml similar to how the gradle build system works.

@SusanRatiLane You should familiarize yourself with [Compiler.java](https://github.com/mit-cml/appinventor-sources/blob/master/appinventor/buildserver/src/com/google/appinventor/buildserver/Compiler.java) before starting this project.

**Comment by ShreyashSaitwal** [comment]
Hi @SusanRatiLane,
Are you still working on this?

**Comment by ShreyashSaitwal** [comment]
> Hi @SusanRatiLane,
> Are you still working on this?

@SusanRatiLane?

**Comment by ewpatton** [comment]
This is going to be fixed by #3292

**Comment by patryk84a** [comment]
> directory structure similar to how Android Archive (.aar) files work

Yes, I think we need to use a technique similar to adding aar libraries here. Because we need to add an R.txt file and a package for a given library that requires an R.class file. I think the R.txt file must contain predefined identifiers like built-in libraries do. But without it aapt2 also indexes resources correctly. I wonder if we should just create the same R.class file that is created in the main application package, in the library package for which we are adding resources.
